### PR TITLE
Issue 763 send funds

### PR DESCRIPTION
**Pull request by Ekaanth**
Send Funds test case

### CODE REVIEW

File: jest/setup.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import 'react-native-gesture-handler/jestSetup';
 
+global.__reanimatedWorkletInit = jest.fn();
```
Comment (silva-fj) [inline jest/setup.js]: This is duplicated, please remove

File: src/testUtils.tsx [hunk 2]
```diff
@@ -35,5 +36,16 @@ jest.mock('@react-navigation/native', () => {
   };
 });
 
+jest.mock('react-native-qrcode-scanner/node_modules/react-native-permissions', () =>
```
Comment (silva-fj) [inline src/testUtils.tsx]: General mocks should live in `jest/setup.js `, please move all of them there.

File: src/ui/screens/Parachains/OverviewScreen.tsx [hunk 3]
```diff
@@ -30,60 +31,54 @@ export function ParachainsOverviewScreen({navigation}: ScreenProps) {
...
-                          <Row label="Remaining">
-                            <Caption>{parachainsSummary.leasePeriod.remainderParts[0]}</Caption>
-                          </Row>
-                        </>
-                      ) : null}
+                    <Padder scale={1} />
+                    <Divider />
+                    <Padder scale={1} />
+                    <View style={styles.itemRow}>
+                      <StatInfoBlock title="Current lease">{parachainsSummary?.leasePeriod.currentLease}</StatInfoBlock>
```
Comment (silva-fj) [inline src/ui/screens/Parachains/OverviewScreen.tsx]: You are also changing the UI in this PR, please provide screenshots so we can see the changes.

File: src/ui/screens/SendFundScreen.test.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+} as unknown as NavigationProp<AccountsStackParamList, typeof sendFundScreen>;
+
+const route = {
+  params: {
+    address: '14yx4vPAACZRhoDQm1dyvXD3QdRQyCRRCe5tj1zPomhhS29a',
+  },
+} as RouteProp<AccountsStackParamList, typeof sendFundScreen>;
+
+describe('SendFundScreen component', () => {
+  it('On component loaded successfully', async () => {
```
Comment (silva-fj) [inline src/ui/screens/SendFundScreen.test.tsx]: Remember that the description is to give an idea of what you would expect on this case, you need to read the description counting the "It" and it should makes sense.. If you read `It on component loaded successfully` you notice that there is something missing  in that sentence.

Another option is to use `test(...)` where you describe what you are testing.

File: src/ui/screens/SendFundScreen.test.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      render(<SendFundScreen navigation={navigation} route={route} />),
+    );
+    fireEvent.changeText(getByPlaceholderText('Enter amount'), '2588');
+    fireEvent.changeText(getByPlaceholderText('Account address'), 'test');
+    expect(queryAllByText('Enter a valid address').length).toEqual(1);
+    fireEvent.changeText(getByPlaceholderText('Account address'), '14yx4vPAACZRhoDQm1dyvXD3QdRQyCRRCe5tj1zPomhhS29a');
+    expect(queryAllByText('Enter a valid address').length).toEqual(0);
+    getByText('Transfer with account keep-alive checks');
+    const button = getByText('Make Transfer');
+    fireEvent.press(button);
```
Comment (silva-fj) [inline src/ui/screens/SendFundScreen.test.tsx]: This you are firing this event, you should test what should happen next.

File: src/ui/screens/SendFundScreen.test.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    fireEvent.changeText(getByPlaceholderText('Account address'), 'test');
+    expect(queryAllByText('Enter a valid address').length).toEqual(1);
+    fireEvent.changeText(getByPlaceholderText('Account address'), '14yx4vPAACZRhoDQm1dyvXD3QdRQyCRRCe5tj1zPomhhS29a');
+    expect(queryAllByText('Enter a valid address').length).toEqual(0);
+    const switchComponent = getByTestId('keep_alive_switch');
+    fireEvent(switchComponent, 'valueChange', false);
+    await waitFor(() => queryByText('Normal transfer without keep-alive checks'));
+    expect(queryByText('Transfer with account keep-alive checks')).toBe(null);
+    const button = getByText('Make Transfer');
+    fireEvent.press(button);
```
Comment (silva-fj) [inline src/ui/screens/SendFundScreen.test.tsx]: Same here

File: src/ui/screens/SendFundScreen.test.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    getAllByText('Enter amount');
+    getAllByText('Send to address');
+    getAllByText('Existential deposit');
+    getAllByText('Transfer with account keep-alive checks');
+    expect(queryByText('Normal transfer without keep-alive checks')).toBe(null);
+    expect(getByText(/Make Transfer/i)).toBeDisabled();
+    getByA11yState({disabled: true});
+  });
+
+  it('When user enter correct values and checked keep alive check', async () => {
```
Comment (silva-fj) [inline src/ui/screens/SendFundScreen.test.tsx]: `"When user enter correct values and checked keep alive check"` then what? remember to add here what you are expecting/testing to make it easier to reason about and have an idea of what this is for without having to read the code.

**Review by silva-fj — COMMENTED** [review]
FYI: I fixed the TS error you had

File: jest/setup.js [hunk 8]
```diff
@@ -14,3 +14,24 @@ global.__reanimatedWorkletInit = jest.fn();
 
 jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
 jest.mock('react-native-mmkv');
+
+const mockedNavigation = jest.fn();
+
+jest.mock('@react-navigation/native', () => {
+  return {
+    useNavigation: () => ({
+      navigate: mockedNavigation,
```
Comment (silva-fj) [inline jest/setup.js]:
```suggestion
      navigate: jest.fn(),
```

File: jest/setup.js [hunk 9]
```diff
@@ -14,3 +14,24 @@ global.__reanimatedWorkletInit = jest.fn();
 
 jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
 jest.mock('react-native-mmkv');
+
+const mockedNavigation = jest.fn();
```
Comment (silva-fj) [inline jest/setup.js]: NIT: If this is not used anywhere else use inline config. If you want to keep it consider changing the name to `navigateMock`

**Review by silva-fj — APPROVED** [review]

**Comment by Ekaanth** [comment]
This PR needs a refactor on the `useApiTx` hook until further discussions this  PR is blocked. ⛔️ @franjs @ishrak-imam

**Comment by ishrak-imam** [comment]
Although this test suit is following the correct approach, just need to test what happens after the `Make transfer` button pressed as @franjs suggested, I think it makes more sense to test the Send fund component after we refactor the Modalize to use BottomSheet.

But that should not be a blocker for this PR.

**Comment by silva-fj** [comment]
@Ekaanth Can you create an issue for the remaining work, add a TODO here in the code pointing to that issue and get this merged?

**Pull request merged**